docs/developer/getting-started/debugging.asciidoc

```diff
@@ -130,3 +130,71 @@ Once you're finished, you can stop Kibana normally, then stop the {es} and APM s
+
+Let's start by defining a problem area you are interested in. For example, you might be interested in seeing how a particular Kibana Plugin is performing, so no need to gather logs for all of Kibana. Or you want to focus on a particular feature, such as requests from the Kibana server to the Elasticsearch server.
+Depending on your needs, you can configure Kibana to generate logs for a specific feature.
+[source,yml]
+----
+logging:
+  appenders:
+    file:
+      type: file
+      fileName: ./kibana.log
+      layout:
+        type: json
+
+### gather all the Kibana logs into a file
+logging.root:
+    appenders: [file]
+    level: all
+
+### or gather a subset of the logs
+logging.loggers:
+  ### responses to an HTTP requests
+  - name: http.server.response
+    level: debug
+    appenders: [file]
+  ### result of a query to the Elasticsearch server
+  - name: elasticsearch.query
+    level: debug
+    appenders: [file]
+  ### logs generated by my plugin
+  - name: plugins.myPlugin
+    level: debug
+    appenders: [file]
+----
+WARNING: Note that we configured Kibana to produce logs in https://www.elastic.co/guide/en/ecs/master/ecs-reference.html[ECS JSON] format. At the moment, it's the only format including meta-information necessary for https://www.elastic.co/guide/en/apm/agent/nodejs/current/log-correlation.html[log correlation] out-of-the-box.
+
+The next step is to define what https://www.elastic.co/observability[observability tools] are available to you. 
+For a better Developer Experience, we recommend you to set up https://www.elastic.co/guide/en/apm/get-started/current/observability-integrations.html[Observability integration] provided by Elastic to debug your application with the powerful <<debugging-logs-apm-ui, APM UI.>>
+If you need to debug something quickly without the necessity to set up additional tooling, you can deal with <<plain-kibana-logs, the plain Kibana logs yourselves.>>
+
+[[debugging-logs-apm-ui]]
+==== APM UI
+*Prerequisites* Kibana logs are configured to be in ECS JSON format to include tracing identifiers.
+
+If you would like to debug Kibana with Kibana APM UI, you need to set up the APM infrastructure. You can find instructions for the setup process
+https://www.elastic.co/guide/en/apm/get-started/current/observability-integrations.html[on this page].
+
+Once you set up the APM infrastructure, you can enable the APM agent and put Kibana under load to collect APM events. To analyze the collected metrics and logs, use the APM UI as demonstrated https://www.elastic.co/guide/en/kibana/master/transactions.html#transaction-trace-sample[in the docs].
+
+[[plain-kibana-logs]]
+==== Plain Kibana Logs
+*Prerequisites* Kibana logs are configured to be in ECS JSON format to include tracing identifiers.
+
+Open Kibana Logs and search for an operation you are interested in.
+For example, we want to investigate the slowness of the `/api/telemetry/v2/clusters/_stats` Kibana endpoint.
```

Comment:
When I read this sentence, it gives me the impression that the `/api/telemetry/v2/clusters/_stats` endpoint is _always_ slow. Maybe we can rephrase this to something more positive, like:
```suggestion
For example, we want to investigate the response times for queries to the `/api/telemetry/v2/clusters/_stats` Kibana endpoint.
```